persistence/modeshape-persistence-relational/src/main/java/org/modeshape/persistence/relational/DefaultStatements.java

```diff
@@ -297,21 +332,20 @@ public void remove( List<String> ids ) throws SQLException {
             runBatchOperation(connection, sql, ids, batchLoadSize(), this::batchRemove);    
         }
 
-        private void batchRemove( Connection connection, String statement, int startIdx, int endIdx, List<String> ids )
-                throws SQLException {
-            List<String> sublist = ids.subList(startIdx, endIdx);
-            String params = sublist.stream().map(id -> "?").collect(Collectors.joining(","));
+        private boolean batchRemove( Connection connection, String statement, List<String> ids ) throws SQLException {
+            String params = ids.stream().map(id -> "?").collect(Collectors.joining(","));
             String statementString = statement.replaceAll("#", params);
```

Comment:
Lines 336 and 337 are used in several places (at least 3). Would it be useful to extract to a helper method? It'd probably help with understanding the code; took me a moment to figure out what they're doing.